llvm/lib/Target/AArch64/AArch64InstrInfo.td

```diff
@@ -6627,6 +6627,17 @@ def : Pat<(f16 (any_uint_to_fp (i32 (any_fp_to_uint f16:$Rn)))),
           (UCVTFv1i16 (f16 (FCVTZUv1f16 f16:$Rn)))>;
 }
 
+let Predicates = [HasNEONandIsSME2p2StreamingSafe] in {
```

Comment:
Is this necessary, it looks like you're already inside a `HasNEONandIsSME2p2StreamingSafe` block?